### PR TITLE
Translate the collection signature article.

### PR DESCRIPTION
**Pull request by Natim**
r? @tarekziade @n1k0 @ametaireau @magopian

### CODE REVIEW

File: content/2015.08.donnees-et-signatures-en.rst [hunk 1]
```diff
@@ -0,0 +1,144 @@
+Sign collection to enforce data integrity
```
Comment (almet) [inline content/2015.08.donnees-et-signatures-en.rst]: s/force/check

Comment (almet) [inline content/2015.08.donnees-et-signatures-en.rst]: ou "ensure".

File: content/2015.08.donnees-et-signatures-en.rst [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+
+:date: 2015-08-26
+:summary: How to enforce data integrity to Kinto users using collection hash signature.
+:slug: data-signature
+:lang: en
+:url: data-signature
+
+.. note::
+
+    Translated from the French by `Rémy Hubscher <https://twitter.com/Natim>`_.
```
Comment (almet) [inline content/2015.08.donnees-et-signatures-en.rst:12]: Je pense qu'il n'est pas necessaire de mettre en avant les compte twitter ici.
Voir discussion sur https://github.com/mozilla-services/servicedenuages.fr/pull/53#discussion_r38112576

File: content/2015.08.donnees-et-signatures-en.rst [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+
+Within the scope of the `Go Faster <https://wiki.mozilla.org/Firefox/Go_Faster>`_
+project, we want to distribute minor Firefox updates more often than
```
Comment (almet) [inline content/2015.08.donnees-et-signatures-en.rst]: s/minor//

File: content/2015.08.donnees-et-signatures-en.rst [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+just with major updates (which happens once every six weeks).
+
+There are multiple kind of data that we will need to update on the client.
+Among others, we would like to `update the SSL OneCRL (Control Revocation List)
```
Comment (almet) [inline content/2015.08.donnees-et-signatures-en.rst]: OneCRL → CRL. Consider "update the SSL Control Revocation List (named OneCRL)"

File: content/2015.08.donnees-et-signatures-en.rst [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+<https://blog.mozilla.org/security/2015/03/03/revoking-intermediate-certificates-introducing-onecrl/>`_.
```
Comment (almet) [inline content/2015.08.donnees-et-signatures-en.rst:20]: If we add this link here we better add it to the original article.

File: content/2015.08.donnees-et-signatures-en.rst [hunk 6]
```diff
@@ -0,0 +1,144 @@
...
+
+In that specific case, it is obviously very important to make sure
+that all the fetched data are coming from Mozilla and that no records
+are missing. So that we can be absolutely confident that nobody added
```
Comment (almet) [inline content/2015.08.donnees-et-signatures-en.rst]: consider: "Then, we can be absolutely"

File: content/2015.08.donnees-et-signatures-en.rst (hunk 5 above)
Comment (Natim) [inline content/2015.08.donnees-et-signatures-en.rst:20]: We don't add it it is already present in the original article.

File: content/2015.08.donnees-et-signatures-en.rst (hunk 2 above)
Comment (Natim) [inline content/2015.08.donnees-et-signatures-en.rst:12]: On n'était pas tous d'accord, du coup je propose que ce soit le traducteur qui choisisse de le faire ou pas.

File: content/2015.08.donnees-et-signatures-en.rst [hunk 7]
```diff
@@ -0,0 +1,144 @@
...
+This mecanism works in our specific use-cases of the OneCRL update
+project, but it can also be useful to any future project that needs to
+make sure that all records on a collection have been correctly synced.
+We will probably need to re-use it to update other part of Firefox or
+Fennec but you may also want to use it for your use-cases.
+
+In order to distribute the data or meta-data associated with the files
+that needs to be updated, we plan to use `Kinto
+<https://kinto.readthedocs.org>`_. Kinto is a good fit because it can
+easily be cached behind a CDN.
```
Comment (ametaireau) [inline content/2015.08.donnees-et-signatures-en.rst]: This isn't part of the original article. I'm not actually sure the fact it's easily cachable makes kinto a good fit here. The fact that it handles synchronisation out the box probably would be a better thing to point out.

File: content/2015.08.donnees-et-signatures-en.rst [hunk 8]
```diff
@@ -0,0 +1,144 @@
...
+
+Also we don't want our user to trust either the CDN or the Kinto
+server itself without checking. Somebody can attack the CDN or the
+Kinto server and may add or remove records and update the CRL in a
+wrong way. If you think about it is a horrible scenario.
```
Comment (almet) [inline content/2015.08.donnees-et-signatures-en.rst]: This is a good addition, though. Can you update the original article with it?

File: content/2015.08.donnees-et-signatures-en.rst [hunk 9]
```diff
@@ -0,0 +1,144 @@
...
+Fennec but you may also want to use it for your use-cases.
+
+In order to distribute the data or meta-data associated with the files
+that needs to be updated, we plan to use `Kinto
+<https://kinto.readthedocs.org>`_. Kinto is a good fit because it can
+easily be cached behind a CDN.
+
+Also we don't want our user to trust either the CDN or the Kinto
+server itself without checking. Somebody can attack the CDN or the
+Kinto server and may add or remove records and update the CRL in a
```
Comment (almet) [inline content/2015.08.donnees-et-signatures-en.rst]: "in a wront way" is redundant with "in a wrong way". I would remove the latter.

File: content/2015.08.donnees-et-signatures-en.rst [hunk 10]
```diff
@@ -0,0 +1,144 @@
...
+wrong way. If you think about it is a horrible scenario.
+
+Consider the following workflow:
+
+- The person responsible to update the CRL, *the updater*, has got a
+  private key (or even better `a HSM
+  <https://en.wikipedia.org/wiki/Hardware_Security_Module>`_ which
+  will enable her to sign a hash of the collection records;
```
Comment (almet) [inline content/2015.08.donnees-et-signatures-en.rst:52]: unbalanced parenthesis

File: content/2015.08.donnees-et-signatures-en.rst [hunk 11]
```diff
@@ -0,0 +1,144 @@
...
+- The corresponding public key ships with the client (Firefox or Fennec);
+- *Hashing* and *signature* are done on client side to prevent some
+  attacks vectors (if somebody can access the Kinto server for
+  instance.)
+
+The signing is a one-way operation called *hashing* that guarantee the
```
Comment (almet) [inline content/2015.08.donnees-et-signatures-en.rst]: signing and hashing are two different things.

File: content/2015.08.donnees-et-signatures-en.rst [hunk 12]
```diff
@@ -0,0 +1,144 @@
...
+same result or the same input.
+
+
+First data setup on Kinto
```
Comment (almet) [inline content/2015.08.donnees-et-signatures-en.rst]: "First issuance of the data on the server"

File: content/2015.08.donnees-et-signatures-en.rst [hunk 13]
```diff
@@ -0,0 +1,144 @@
...
+=========================
+
+All the data are fetched from a *secured* source and converted into a
+JSON collection. Each record have got a unique ID generated on client side.
```
Comment (almet) [inline content/2015.08.donnees-et-signatures-en.rst]: consider: "each record is assigned an unique ID"

File: content/2015.08.donnees-et-signatures-en.rst [hunk 14]
```diff
@@ -0,0 +1,144 @@
...
+
+For instance, we could have the following record:
+
+.. code-block:: javascript
+
+   {"id": "b7dded96-8df0-8af8-449a-8bc47f71b4c4",
+    "fingerprint": "11:D5:D2:0A:9A:F8:D9:FC:23:6E:5C:5C:30:EC:AF:68:F5:68:FB:A3"}
+
+Then the collection *hash* is digested, signed and then send to the Kinto server.
```
Comment (almet) [inline content/2015.08.donnees-et-signatures-en.rst]: I'm not sure "digested" is the right term here. consider "computed".

Comment (almet) [inline content/2015.08.donnees-et-signatures-en.rst]: s/send/sent

File: content/2015.08.donnees-et-signatures-en.rst [hunk 15]
```diff
@@ -0,0 +1,144 @@
...
+(See below for details)
+
+The signature process is deported to a specific service in a sandbox
+that make sure of the certificate security which is crucial in the
```
Comment (almet) [inline content/2015.08.donnees-et-signatures-en.rst]: that ensures the certificate security

File: content/2015.08.donnees-et-signatures-en.rst [hunk 16]
```diff
@@ -0,0 +1,144 @@
...
+How to validate data integrity?
+===============================
+
+First, we need to fetch the collection records as well as the hash and
+the signature.
+
+Then, we can validate that the hash signature is valid to make sure it
+has been generated by a trusted source.
+
+Finally we can digest our local collection hash to make sure it
```
Comment (almet) [inline content/2015.08.donnees-et-signatures-en.rst]: I would use "serialize".

File: content/2015.08.donnees-et-signatures-en.rst [hunk 17]
```diff
@@ -0,0 +1,144 @@
...
+matches the signed one.
+
+
+Update the collection data
+==========================
+
+When you need to `CRUD
```
Comment (almet) [inline content/2015.08.donnees-et-signatures-en.rst]: I doubt CRUD is a verb ;) "edit" could work and conveys enough meaning to be useful I believe.

File: content/2015.08.donnees-et-signatures-en.rst [hunk 18]
```diff
@@ -0,0 +1,144 @@
...
+<https://en.wikipedia.org/wiki/Create,_read,_update_and_delete>`_
+records of the collection, we need to make sure that our local
```
Comment (almet) [inline content/2015.08.donnees-et-signatures-en.rst]: s/we/the client

Comment (almet) [inline content/2015.08.donnees-et-signatures-en.rst]: s/our/the

File: content/2015.08.donnees-et-signatures-en.rst [hunk 19]
```diff
@@ -0,0 +1,144 @@
...
+collection records matches the remove server ones and are valid.
+
+Once we are confident that our collection update is valid, we can
+digest the new collection hash and sign it.
```
Comment (almet) [inline content/2015.08.donnees-et-signatures-en.rst]: Same for "we" and "our".

**Comment by almet** [comment]
Cool! Good trad overall :-) I've added a few nits and comments.

File: content/2015.08.donnees-et-signatures-en.rst (hunk 7 above)
Comment (Natim) [inline content/2015.08.donnees-et-signatures-en.rst]: En fait c'était pour donner un cas encore plus concret de pourquoi on pouvait pas seulement faire confiance au serveur

File: content/2015.08.donnees-et-signatures-en.rst (hunk 14 above)
Comment (Natim) [inline content/2015.08.donnees-et-signatures-en.rst]: isn't it `hashlib.sha256('blob').digest()`

File: content/2015.08.donnees-et-signatures-en.rst (hunk 17 above)
Comment (Natim) [inline content/2015.08.donnees-et-signatures-en.rst]: CRUD stand for Create, Read, Update or Delete, but I can expand it :)

File: content/2015.08.donnees-et-signatures-en.rst (hunk 2 above)
Comment (almet) [inline content/2015.08.donnees-et-signatures-en.rst:12]: On va ajouter ça au theme avec un champ "traductor".

**Pull request merged**